Issue comment event handler

lmeerkatz opened the pull request:
Methods and tests to handle lurch commands posted as comments on issues. 

Commands are:
* add // adds an issue from github as a work item in AA
* attach: [work name] // attaches an issue in github to an existing work record in AA
* remove // removes all links to a given github issue from any attached record in AA

@dextermilo

kbromer commented:
Was just starting to play with this, but I think the pull request/branch is missing some required dependent objects: 
Github_Repositories__c
Github_Request__c
Github_Link__c

ll Component Failures:
1.  classes/GithubIssueCommentHandler.cls -- Error: Invalid type: GithubToAgileAccelerator.Action (line 25, column 13)
2.  classes/GithubIssueCommentHandler_Mock.cls -- Error: Invalid type: Github_Repositories__c (line 5, column 20)
3.  classes/GithubIssueCommentHandler_Test.cls -- Error: Dependent class is invalid and needs recompilation:
    GithubIssueCommentHandler: line 25, column 13: Invalid type: GithubToAgileAccelerator.Action (line 5, column 3)
4.  classes/GithubRequestHandler.cls -- Error: sObject type 'Github_Request__c' is not supported. If you are attempting to use a custom object, be sure to append the '__c' after the entity name. Please reference your WSDL or the describe call for the appropriate names. (line 17, column 13)
5.  classes/GithubToAgileAccelerator.cls -- Error: sObject type 'GitHub_Link__c' is not supported. If you are attempting to use a custom object, be sure to append the '__c' after the entity name. Please reference your WSDL or the describe call for the appropriate names. (line 142, column 36)
6.  classes/GithubToAgileAccelerator_Test.cls -- Error: sObject type 'Github_Request__c' is not supported. If you are attempting to use a custom object, be sure to append the '__c' after the entity name. Please reference your WSDL or the describe call for the appropriate names. (line 30, column 44)
7.  triggers/GithubRequest_Trigger.trigger -- Error: Invalid SObject type name: Github_Request__c (line 1, column 34)

lmeerkatz commented:
@kbromer Here you go. Added:
GitHub_Link__c.object
Github_Integration_Users__c.object
Github_Repositories__c.object
Github_Request__c.object

kbromer commented:
1.  objects/Github_Request__c.object (Github_Request__c.All) -- Error: Not in package.xml
2.  classes/GithubIssueCommentHandler_Test.cls -- Error: Invalid type: GithubIssueCommentHandler.CommentAction (line 5, column 3)

The second one is an ApexAction, right?  

Any idea what the first one is?

kbromer commented:
Weird - 1 is a listview i think... is there a dependency on the listview somewhere?

lmeerkatz commented:
Not sure what the deal is the with list view, but I tried removing it from the object to see if that helps. 
I updated the test to fix the CommentAction issue. I had refactored a bunch of stuff but forgot about that test class.

kbromer commented:
@lmeerkatz okay - all tests pass, we've over 95% coverage on each class.  

Feel free to deploy and give it a shot - alternately, this repo is wired up to the .kbromer sandbox, just recently cut - you can test out the behaviors there. If you're good on it, I'd like to get this merged and then deployed to prod as soon as is reasonable

The following commands should work: 
```
**lurch:add 
**lurch:attach W-123456
**lurch:detach
```
They should be effective in issues, issue comments and pull requests

Once you create issues, you can flush the queue by running the batch job directly from the dev console: 
```
GithubRequestHandler ghr = new GithubRequestHandler();
ghr.processRequests();
```

That should create an issue, add a chatter comment to an existing issue, detach a github link from an issue, etc. as needed. It should also post back to Github for any add, attach or detach command.

njjc commented:
Amazing news!

On Wed, Sep 9, 2015 at 7:52 PM, Kevin Bromer <notifications@github.com>
wrote:

> @lmeerkatz <https://github.com/lmeerkatz> okay - all tests pass, we've
> over 95% coverage on each class.
>
> Feel free to deploy and give it a shot - alternately, this repo is wired
> up to the .kbromer sandbox, just recently cut - you can test out the
> behaviors there. If you're good on it, I'd like to get this merged and then
> deployed to prod as soon as is reasonable
>
> The following commands should work:
>
> **lurch:add
> **lurch:attach W-123456
> **lurch:detach
>
> They should be effective in issues, issue comments and pull requests
>
> Once you create issues, you can flush the queue by running the batch job
> directly from the dev console:
>
> GithubRequestHandler ghr = new GithubRequestHandler();
> ghr.processRequests();
>
> That should create an issue, add a chatter comment to an existing issue,
> detach a github link from an issue, etc. as needed. It should also post
> back to Github for any add, attach or detach command.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SalesforceFoundation/sfdo-github/pull/1#issuecomment-139101481>
> .
>

kbromer commented:
Actually if you have a few minutes @njjc, would love you to check it out in the SB too!  You'll need to add your Github UN to the custom setting, but it should be good to go after that-

njjc commented:
Definitely, what's the sandbox suffix?

kbromer commented:
.kbromer

njjc commented:
I'm going to need help getting setup perms.

On Thu, Sep 10, 2015 at 11:10 AM, Kevin Bromer <notifications@github.com>
wrote:

> .kbromer
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SalesforceFoundation/sfdo-github/pull/1#issuecomment-139330845>
> .
>

LurchTheButler commented:
Tracking <a href="https://foundation--kbromer.cs42.my.salesforce.com/a2x56000000572NAAQ" target="_blank">W-008057</a>

lmeerkatz commented on src/classes/GithubRequestHandler.cls:
```diff
@@ -0,0 +1,136 @@
+            requestIds.add(gr.Github_Unique_Id__c);
+          }
+
+          List<Github_Link__c> igl = new List<Github_Link__c>();
```
What does igl mean?

kbromer replied:
individual github links maybe? 

Something like that...

kbromer commented:
@lmeerkatz removed a couple of unused fields, one from GitHub_Link__c and one from the Github_Request__c object.  Tweaked the package.xml as well to be easier to read. 

This should be good to go per our conversation earlier today - thx!
